src/main/java/org/joget/marketplace/TimeAgoDatalistFormatter.java

```diff
@@ -24,103 +23,118 @@ public String getVersion() {
     }
 
+    public String getYear() {
+        // support i18n
+        return " " + AppPluginUtil.getMessage("org.joget.marketplace.TimeAgoDatalistFormatter.year(s)", getClassName(), MESSAGE_PATH);
+    }
+
+    public String getMonth() {
+        // support i18n
+        return " " + AppPluginUtil.getMessage("org.joget.marketplace.TimeAgoDatalistFormatter.month(s)", getClassName(), MESSAGE_PATH);
+    }
+
+    public String getDay() {
+        // support i18n
+        return " " + AppPluginUtil.getMessage("org.joget.marketplace.TimeAgoDatalistFormatter.day(s)", getClassName(), MESSAGE_PATH);
+    }
+
     @Override
     public String format(DataList dataList, DataListColumn column, Object row, Object value) {
-        
+
         AppDefinition appDef = AppUtil.getCurrentAppDefinition();
         String result = (String) value;
+
         String duration = getPropertyString("duration");
-        
-        try {
-            Period dateDiff;
-        
-            if (duration.equals("today")) {
-                if(result.isEmpty()){
-                    return result;
-                }
-                LocalDate date = LocalDate.parse(result); //Date
-                LocalDate currentDate = LocalDate.now(); //Current Date
+        Period dateDiff;
+
+        if (duration.equals("today")) {
+
+            LocalDate date = LocalDate.parse(result); // Date
+            LocalDate currentDate = LocalDate.now(); // Current Date
+
+            // Duration From Column Date To Today
+            try {
 
-                //Duration From Column Date To Today
                 dateDiff = Period.between(currentDate, date);
 
                 if (Math.abs(dateDiff.getYears()) > 0) {
-                    return Math.abs(dateDiff.getYears()) + " year(s) " + Math.abs(dateDiff.getMonths()) + 
-                            " month(s) " + Math.abs(dateDiff.getDays()) + " day(s)";
+                    return Math.abs(dateDiff.getYears()) + getYear() + Math.abs(dateDiff.getMonths()) +
+                            getMonth() + Math.abs(dateDiff.getDays()) + getDay();
                 } else if (Math.abs(dateDiff.getMonths()) > 0) {
-                    return Math.abs(dateDiff.getMonths()) + " month(s) " + Math.abs(dateDiff.getDays()) + " day(s)";
+                    return Math.abs(dateDiff.getMonths()) + getMonth() + Math.abs(dateDiff.getDays()) + getDay();
                 } else {
-                    return Math.abs(dateDiff.getDays()) + " day(s)";
+                    return Math.abs(dateDiff.getDays()) + getDay();
                 }
 
-            } else if (duration.equals("anotherDate")) {
-                if(result.isEmpty()){
-                    return result;
-                }
-                LocalDate date = LocalDate.parse(result); //Date
+            } catch (Exception e) {
+               LogUtil.error("org.sample.TimeAgoDatalistFormatter", e, "Error!!");
```

Comment:
org.sample.TimeAgo is not correct.

Reply:
I have a question here Should it be the same as this one
 LogUtil.error(TimeAgoDatalistFormatter.class.getName(), e, "Not able to compute duration"); 
?

Reply:
yup... yes please.